superset/daos/chart.py

```diff
@@ -41,16 +41,14 @@ class ChartDAO(BaseDAO[Slice]):
 
     @classmethod
     def delete(cls, items: Slice | list[Slice], commit: bool = True) -> None:
-        item_ids = [item.id for item in get_iterable(items)]
```

Comment:
@jfrag1 if we're removing the bulk delete logic, i.e., deleting outside of the ORM, then I think there's merit in removing the entire function and falling back to `BaseDAO.delete`.

Reply:
I didn't do this here because of this logic which is specific to charts:
```
for item in get_iterable(items):
    item.dashboards = []
    db.session.merge(item)
```

Reply:
@jfrag1 the reason this logic is required is because `ON DELETE CASCADE` isn't configured. The right solution would be (in a separate PR) to replicate the logic outlined in https://github.com/apache/superset/pull/24628 for the `dashboard_slices` table.

Reply:
@jfrag1 I authored https://github.com/apache/superset/pull/24938 which should aid with my suggestion above.